### PR TITLE
[BACKLOG-25403] Remove the javax.ws.rs-api dependency

### PR DESCRIPTION
**Pull request by joaormf**
@tgf @marcoslarsen @JoaoRPF @plagoa @graimundo please review

### CODE REVIEW

**Comment by joaormf** [comment]
Remove the javax.ws.rs-api dependency after https://jira.pentaho.com/browse/COMP-8856 alert that it a new component registration was needed because that was a new dependency.

File: core/src/main/java/pt/webdetails/cda/utils/QueryParameters.java [hunk 1]
```diff
@@ -104,4 +113,50 @@ private static Object getParam( List<String> paramValues ) {
...
+        } else {
+          ArrayList<V> newValuesList = new ArrayList<>();
+          newValuesList.add( value );
+          super.put( key, newValuesList );
+        }
+      }
+    }
+
+    @Override public void putSingle( K key, V value ) {
+      throw new UnsupportedOperationException(  );
```
Comment (tgf) [inline core/src/main/java/pt/webdetails/cda/utils/QueryParameters.java]: why not keep it abstract?

File: core/src/main/java/pt/webdetails/cda/utils/QueryParameters.java [hunk 2]
```diff
@@ -104,4 +113,50 @@ private static Object getParam( List<String> paramValues ) {
...
     return paramValues;
   }
+
+  /**
+   * Local implementation of the MultivaluedHashMap that is bundled on a javax ws rs api dependency.
+   * It's custom implemented here to avoid bringing a dependency just for the simple use of this.
+   * @param <K> The keys type.
+   * @param <V> The values type.
+   */
+  protected class MultiValueQueryParameter<K, V> extends HashMap<K, List<V>> implements MultivaluedMap<K, V> {
```
Comment (tgf) [inline core/src/main/java/pt/webdetails/cda/utils/QueryParameters.java]: why implement the interface if we dont need it? can't we just use a Map<String,List<String>>?

Comment (joaormf) [reply core/src/main/java/pt/webdetails/cda/utils/QueryParameters.java]: Excelent point

File: core/src/main/java/pt/webdetails/cda/push/WebsocketJsonQueryEndpoint.java [hunk 3]
```diff
@@ -56,8 +53,8 @@
   @Override
   public void onMessage( String message, Consumer<String> outboundMessageConsumer ) {
     try {
-      MultivaluedMap<String, String> params = getParameters( message );
-      DoQueryParameters parameters = getDoQueryParameters( params );
+      QueryParameters.MultiValueQueryParameter<String, String> params = QueryParameters.getInstance().getParameters( message );
+      DoQueryParameters parameters = QueryParameters.getInstance().getDoQueryParameters( params );
```
Comment (graimundo) [inline core/src/main/java/pt/webdetails/cda/push/WebsocketJsonQueryEndpoint.java]: This still is a static reference ( `QueryParameters.getInstance()` ) so you didn't actually do any decoupling here. Use a getter/setter where the QueryParameters can be injected or a class attribute with construction injection. If you don't want to go through the trouble of injecting it in Spring / Blueprint you can have a default constructor that uses QueryParamenters.getInstance() as the default value for the property.

File: core/src/main/java/pt/webdetails/cda/utils/QueryParameters.java [hunk 4]
```diff
@@ -104,4 +112,26 @@ private static Object getParam( List<String> paramValues ) {
...
     return paramValues;
   }
+
+  /**
+   * Local implementation of the MultivaluedHashMap that is bundled on a javax ws rs api dependency.
+   * It's custom implemented here to avoid bringing a dependency just for the simple use of this.
+   * @param <K> The keys type.
+   * @param <V> The values type.
+   */
+  public static class MultiValueQueryParameter<K, V> extends HashMap<K, List<V>> {
```
Comment (graimundo) [inline core/src/main/java/pt/webdetails/cda/utils/QueryParameters.java]: @joaormf final?

**Pull request merged**